### ISSUE TITLE
watch doesn’t show reference variables.

### ISSUE THREAD

**Issue by phoityne** [phoityne/hdx4vs#1]
Visual Studio does not send EvaluateRequest when adding local variable to watch, 

![vs](https://raw.githubusercontent.com/phoityne/hdx4vs/master/docs/issues/001/issue_vs.gif)
- when adding watch.  
  VS sends ScopesRequest and VariablesRequest.
```
[REQUEST] ScopesRequest {seqScopesRequest = 15, typeScopesRequest = "request", commandScopesRequest = "scopes", argumentsScopesRequest = ScopesArguments {frameIdScopesArguments = 0}}
[RESPONSE]{"seq":36,"type":"response","request_seq":15,"success":true,"command":"scopes","message":"","body":{"scopes":[{"name":"GHCi Scope","variablesReference":1,"namedVariables":null,"indexedVariables":null,"expensive":false}]}}
[REQUEST] VariablesRequest {seqVariablesRequest = 16, typeVariablesRequest = "request", commandVariablesRequest = "variables", argumentsVariablesRequest = VariablesArguments {variablesReferenceVariablesArguments = 1}}
[RESPONSE]{"seq":37,"type":"response","request_seq":16,"success":true,"command":"variables","message":"","body":{"variables":[{"name":"_result","value":"_","type":"IO ()","presentationHint":null,"evaluateName":"_result","variablesReference":0,"namedVariables":null,"indexedVariables":null},{"name":"foo","value":"_","type":"Foo","presentationHint":null,"evaluateName":"foo","variablesReference":0,"namedVariables":null,"indexedVariables":null},{"name":"bar","value":"_","type":"Bar","presentationHint":null,"evaluateName":"bar","variablesReference":0,"namedVariables":null,"indexedVariables":null},{"name":"zig","value":"_","type":"Zig","presentationHint":null,"evaluateName":"zig","variablesReference":0,"namedVariables":null,"indexedVariables":null},{"name":"zag","value":"_","type":"Zag","presentationHint":null,"evaluateName":"zag","variablesReference":0,"namedVariables":null,"indexedVariables":null},{"name":"hoge","value":"_","type":"(Float, [Char], Int)","presentationHint":null,"evaluateName":"hoge","variablesReference":0,"namedVariables":null,"indexedVariables":null},{"name":"huga","value":"_","type":"[Int]","presentationHint":null,"evaluateName":"huga","variablesReference":0,"namedVariables":null,"indexedVariables":null},{"name":"func","value":"function :: Int -> Int -> Int","type":"Int -> Int -> Int","presentationHint":null,"evaluateName":"func","variablesReference":0,"namedVariables":null,"indexedVariables":null},{"name":"fizz","value":"_","type":"FizzBuzz","presentationHint":null,"evaluateName":"fizz","variablesReference":0,"namedVariables":null,"indexedVariables":null},{"name":"buzz","value":"_","type":"FizzBuzz","presentationHint":null,"evaluateName":"buzz","variablesReference":0,"namedVariables":null,"indexedVariables":null},{"name":"fuga","value":"_","type":"Map [Char] FizzBuzz","presentationHint":null,"evaluateName":"fuga","variablesReference":0,"namedVariables":null,"indexedVariables":null}]}}
```

but VSCode does.
![vs](https://raw.githubusercontent.com/phoityne/hdx4vs/master/docs/issues/001/issue_vsc.gif)
- when adding watch.  
  VSCode sends EvaluateRequest.
```
[REQUEST] EvaluateRequest {seqEvaluateRequest = 17, typeEvaluateRequest = "request", commandEvaluateRequest = "evaluate", argumentsEvaluateRequest = EvaluateArguments {expressionEvaluateArguments = "zag", frameIdEvaluateArguments = Just 0, contextEvaluateArguments = Just "watch"}}
[RESPONSE]{"seq":38,"type":"response","request_seq":17,"success":true,"command":"evaluate","message":"","body":{"result":"Zag :: String -> Int -> Zag","type":"Zag","presentationHint":null,"variablesReference":3,"namedVariables":null,"indexedVariables":null}}
```

- when opening tree node.  
    VSCode sends VariablesRequest.

```
[REQUEST] VariablesRequest {seqVariablesRequest = 18, typeVariablesRequest = "request", commandVariablesRequest = "variables", argumentsVariablesRequest = VariablesArguments {variablesReferenceVariablesArguments = 3}}
[RESPONSE]{"seq":39,"type":"response","request_seq":18,"success":true,"command":"variables","message":"","body":{"variables":[{"name":"nameZag","value":"\"zag\"","type":"String","presentationHint":null,"evaluateName":"zag","variablesReference":0,"namedVariables":null,"indexedVariables":null},{"name":"yearZag","value":"2017","type":"Int","presentationHint":null,"evaluateName":"zag","variablesReference":0,"namedVariables":null,"indexedVariables":null}]}}

```

**Comment by phoityne** [comment]
going to use reference attribute of VariablesRequest for the local variables.
as a result, all information (including child) of the inspected variable is shown in the local variable tree view.  

 
![vs_modified](https://raw.githubusercontent.com/phoityne/hdx4vs/master/docs/issues/001/issue_vs_modified.gif)

**Comment by phoityne** [comment]
fixed 0.0.6.0.
